src/utils/json-data.js

```diff
@@ -0,0 +1,135 @@
+const fs = require('fs-extra');
+const path = require('path');
+const reserved = require('reserved-words');
+const shared = require('./shared');
+const glob = require('glob');
+const BUILD_TIME_BINDING = '@buildtime:';
+// Matches {{ stache.jsonData. }} with any number of spaces between the '{{' and 'stache.',
+// any keys following the 'jsonData.', and the closing '}}'
+const angularBindingRegExp = new RegExp(/\{\{\s*stache.jsonData.*?\}\}/g);
+const buildTimeBindingRegExp = new RegExp(`\\{\\{\\s*${BUILD_TIME_BINDING}stache.jsonData.*?\\}\\}`, 'g');
+
+let _globalData;
+
+const getGlobalData = () => {
+  if (!_globalData) {
+    _globalData = buildGlobalDataFromJson();
+  }
+
+  return _globalData;
+};
+
+const buildGlobalDataFromJson = () => {
+  const root = shared.resolveAssetsPath('data');
+  const filePaths = glob.sync(path.join(root, '*.json'));
+
+  if (!filePaths.length) {
+    return;
+  }
+
+  const dataObject = filePaths.reduce((acc, filePath) => {
+    const fileName = path.basename(filePath);
+    const propertyName = convertFileNameToObjectPropertyName(fileName);
+
+    if (!isPropertyNameValid(propertyName)) {
+      console.error(
+        new shared.StachePluginError(
+          `A valid Object property could not be determined from file ${fileName}!
+          The property key '${propertyName}' cannot be used. Please choose another file name.`
```

Comment:
I like the readability of breaking the message into two lines, but it's inadvertently adding a bunch of tab characters to the output. This might be a better compromise:

```
new shared.StachePluginError([
  `A valid Object property could not be determined from file ${fileName}!`,
  `The property key '${propertyName}' cannot be used. Please choose another file name.`
].join(' '));
```